src/slices/blogSlice/index.ts

```diff
@@ -0,0 +1,102 @@
+import { Blog, BlogsState } from './../../types/index';
+import { createSlice,  createAsyncThunk, PayloadAction} from '@reduxjs/toolkit';
+
+export const fetchBlogs = createAsyncThunk<Blog[],  ({page:number, limit:number})>(
+  'todos/fetchBlogs',
+  async function (ApiData) {
+    try {
+      const {page, limit} = ApiData;
+      const response = await fetch(`https://jsonplaceholder.typicode.com/photos?_page=${page}&_limit=${limit}`)
+      const data = await response.json();
+      return data
+    }
+    catch {
+     return null
+    }
+  }
+);
+
+export const fetchAllBlogs = createAsyncThunk<Blog[]>(
+  'todos/fetchAllBlogs',
+  async function () {
+    try {
+      const response = await fetch(`https://jsonplaceholder.typicode.com/photos`)
+      const data = await response.json();
+      return data; 
+    }
+    catch {
+      return null
+    }
+  }
+);
+
+const initialState:BlogsState = {
+  list: [],
+  fullArrayLength: [],
+  limit: 8,
+  page: 1,
+  currentStart: 0,
+  currentEnd: 0,
+  show: '',
+  loading: false,
+  error: null,
+}
+
+const blogSlice = createSlice({
+  name: 'blogs',
+  initialState,
+  reducers: {
+    chooseLimit: (state, action:PayloadAction<number>) => {
+      state.limit = action.payload
+    },
+    nextPage: (state) => {
+      state.list = [];
+      state.page ++ 
+      state.currentStart += state.limit 
```

Comment:
you not really need keep this in store since you can easy calculate it: `currentStart = page * limit`, `currentEnd = (page + 1) * limit`